### PR TITLE
Feat/implement stellar wp asset

### PR DESCRIPTION
**Pull request by dpanta94**
### 🎫 Ticket

[SL-246]

### 🗒️ Description

Overwrittes: https://github.com/stellarwp/assets/blob/main/src/Assets/Asset.php#953
In order to apply: https://github.com/the-events-calendar/tribe-common/blob/30039b85b1ff61af52b66c33f27581decb4ee2f6/src/Tribe/Assets.php#L359

This is necessary to be functioning as expected in installs with symlinks.

### ✔️ Checklist
- [x] Ran `npm run changelog` to add changelog file(s). More info [here](https://docs.theeventscalendar.com/developer/git/changelogs/#process)
- [x] Code is covered by **NEW** `wpunit` or `integration` tests.
- [x] Code is covered by **EXISTING** `wpunit` or `integration` tests.
- [x] Are all the **required** tests passing?
- [x] Automated code review comments are addressed.
- [ ] Have you added Artifacts?
- [x] Check the base branch for your PR.
- [ ] Add your PR to the project board for the release.

[SL-246]: https://stellarwp.atlassian.net/browse/SL-246?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/wpunit/Common/Libraries/__snapshots__/Uplink_Controller_Test__it_should_setup_license_fields__0.snapshot.html [hunk 1]
```diff
@@ -0,0 +1,29 @@
+
+	<div class="stellarwp-uplink__license-field">
```
Comment (lucatume) [inline tests/wpunit/Common/Libraries/__snapshots__/Uplink_Controller_Test__it_should_setup_license_fields__0.snapshot.html:2]: Why this file? Was it originally missing?

File: tests/wpunit/Common/Asset_Test.php [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+				Assets::init()->get( 'stellar-test-' . $i . '-style' )->get_url()
+			);
+		}
+	}
+
+	/**
+	 * @test
+	 */
+	public function it_should_not_return_the_same_thing_when_symlinks() {
+		// The test is that the plugins directory is actually in the path ABSPATH . 'foo/'.
```
Comment (lucatume) [inline tests/wpunit/Common/Asset_Test.php:100]: Move this comment to the test method doc-block?

File: tests/wpunit/Common/Asset_Test.php [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+		self::$back_up = [
+			'hook_prefix' => Config::get_hook_prefix(),
+			'version'     => Config::get_version(),
+			'path'        => Config::get_path(),
+			'relative'    => Config::get_relative_asset_path(),
+		];
+
+		Config::set_hook_prefix( 'bork' );
+		Config::set_version( '1.0.0' );
+		Config::set_path( dirname( __DIR__, 3 ) );
```
Comment (lucatume) [inline tests/wpunit/Common/Asset_Test.php:38]: Use `codecept_root_dir` here? If I understand this correctly, this is to get to Common root.

File: tests/wpunit/Common/Asset_Test.php [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+		// The test is that the plugins directory is actually in the path ABSPATH . 'foo/'.
+		// But in the location ABSPATH . 'wp-content/' user has created a symlink named plugins which points to ABSPATH . 'foo/'.
+		// Our implementation should return a URL inside the wp-content directory while the asset one should return the true URL outside the plugins directory.
+		Config::set_path( ABSPATH . 'foo/the-events-calendar/common/' );
+		$this->set_fn_return( 'dirname', static fn( $dir, $level = 1 ) => $level !== 4 ? $dir : ABSPATH . 'foo/', true );
+
+		Assets::init();
+
+		// Add assets.
+		array_map( [ $this, 'add_assets' ], range( 1, 3 ) );
```
Comment (lucatume) [inline tests/wpunit/Common/Asset_Test.php:109]: I would unroll this for clarity and readability, with a cardinality of 3, I think what is being done would be better conveyed by 3 explicit calls to `add_assets`.
If you're doing that, I would also give the assets more readable names like `feature-base`, `feature-editor` and `feature-frontend`. Tests can provide an API usage example people can be linked to when readable enough.

File: tests/wpunit/Common/Asset_Test.php [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+		foreach ( range( 1, 3 ) as $i ) {
```
Comment (lucatume) [inline tests/wpunit/Common/Asset_Test.php:111]: On the same note, I would unroll this as well. The dozen of assertions with laid out asset names (see my comments) above would improve readability.

**Review by lucatume — CHANGES_REQUESTED** [review]
Feature-wise all fine, but I think the readability of tests could be improved.

File: tests/wpunit/Common/Libraries/__snapshots__/Uplink_Controller_Test__it_should_setup_license_fields__0.snapshot.html (hunk 1 above)
Comment (dpanta94) [reply tests/wpunit/Common/Libraries/__snapshots__/Uplink_Controller_Test__it_should_setup_license_fields__0.snapshot.html:2]: Yes it was and it seems like we need it.

**Review by lucatume — CHANGES_REQUESTED** [review]
Really small final touch.

**Pull request merged**